src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseService.java

```diff
@@ -59,6 +60,22 @@ public CaseDetails createNewCaseDetails(String caseTypeId, String jurisdictionId
         return caseDetails;
     }
 
+    /**
+     * @param content Data received from the client.
+     * @param jurisdictionId of the case.
+     * @return <code>Optional&lt;CaseDetails&gt;<code/> - CaseDetails wrapped in Optional
+     */
+    public Optional<CaseDetails> populateCurrentCaseDetailsWithUserInputs(CaseDataContent content, String jurisdictionId) {
```

Comment:
Can you rename this method to `populateCurrentCaseDetailsWithEventFields` please